p4-16/spec/P4-16-spec.mdk

```diff
@@ -5228,6 +5228,41 @@ types (e.g. `structs`) are copied recursively, and all components
 of `header`s are copied, including "validity" bits. Assignment is
 not defined for `extern` values.
 
+### Compound assignment { #sec-compound-assignment }
+
+Compound-assignment operators provide a shorter syntax for assigning
+the result of an arithmetic or bitwise operator. The compound assignment
+operator expressions have the form `LHS op RHS`, where `op` is one of
+`+=`, `-=`, `*=`, `/=`, `%=`, `&=`, `|=`, `^=`, `<<=`, `>>=`, `|+|=`,
+`|-|=`, and `LHS`, `RHS` are expressions with arithmetic types. The
+following table shows all P4 assignment operators and their equivalent.
+
+|----------|-----------------------------------|------------|---------------|
+| Operator | Operator name                     | Example    | Equivalent of |
++----------+:---------------------------------:+:----------:+:-------------:+
+| `=`      | basic assignment                  | `a = b`    | NA            |
+| `+=`     | addition assignment               | `a += b`   | `a = a + b`   |
+| `-=`     | subtraction assignment            | `a -= b`   | `a = a - b`   |
+| `*=`     | multiplication assignment         | `a *= b`   | `a = a * b`   |
+| `/=`     | division assignment               | `a /= b`   | `a = a / b`   |
+| `%=`     | modulo assignment                 | `a %= b`   | `a = a % b`   |
+| `&=`     | bitwise AND assignment            | `a &= b`   | `a = a & b`   |
+| `|=`     | bitwise OR assignment             | `a |= b`   | `a = a | b`   |
+| `^=`     | bitwise XOR assignment            | `a ^= b`   | `a = a ^ b`   |
+| `<<=`    | bitwise left shift assignment     | `a <<= b`  | `a = a << b`  |
+| `>>=`    | bitwise right shift assignment    | `a >>= b`  | `a = a >> b`  |
+| `|+|=`   | saturating addition assignment    | `a |+|= b` | `a = a |+| b` |
+| `|-|=`   | saturating subtraction assignment | `a |-|= b` | `a = a |-| b` |
+|----------|-----------------------------------|----------- |---------------|
+
+The behavior of every builtin compound-assignment expression with the form `E1 op= E2`,
+where `E1` is a modifiable l-value expression and `E2` is an r-value expression,
+is exactly the same as the behavior of the expression `E1 = E1 op E2`.
+
+The side effect of updating `E1` is sequenced after the value computations, but not
+the side effects of `E1` and `E2` themselves and the evaluations of the operands are,
+as usual, unsequenced relative to each other.
```

Comment:
The current spec actually defines left-to-right evaluation of side effects, in Section 8.1 "Expression evaluation order": https://p4.org/p4-spec/docs/P4-16-v1.2.3.html#sec-expr-eval-order

Reply:
Thank Andy! I have updated the pull request to mention about the description of compound assignment expressions in Section 8.1 "Expression evaluation order".